x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/MergeOperator.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.operator;
+
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.compute.data.BlockFactory;
+import org.elasticsearch.compute.data.Page;
+import org.elasticsearch.core.Releasables;
+
+import java.util.List;
+import java.util.ListIterator;
+import java.util.function.Supplier;
+
+/**
+ * A merge operator is effectively a "fan-in" operator - accepts input
+ * from several sources and provides it in a single output.
+ */
+public class MergeOperator extends SourceOperator {
+
+    private final BlockFactory blockFactory;
+    private boolean finished;
+    private ListIterator<Block[]> subPlanBlocks;
```

Comment:
Could you put these below the MergeOperatorFactory with the rest of the properties?

Reply:
I just get confused when they are split.

Reply:
Done. Since this is a source operator, we can simplify the state quite a bit [a7a23b5](https://github.com/elastic/elasticsearch/pull/121948/commits/a7a23b5195e756001ff03c16e019874ed5fcac8e)